scripts/simplify_network.py

```diff
@@ -743,6 +746,182 @@ def drop_isolated_nodes(n, threshold):
     return n
 
 
+def find_isolated_sub_networks(n, threshold):
```

Comment:
Agree, but the main point of `find_isolated_sub_networks()` is to deal with multi-country sub-networks, while the "main" function deals with finding the backbone networks for each of the countries. Not sure we make the code more clear if moving into the main code also dealing with isolated subnetworks.

Probably, it would be great to have some more understanding on use-cases for this functionality and adjust the code accordingly. Currently, the code has been tested for `["RW", "BI", "CD"]` which is rather tricky which also could have contribute into some over-complication 🙃 

Have added some comments to `find_isolated_sub_networks()` to facilitate further revisions.